### ISSUE TITLE
Support pkg_resources-style namespace packages

### ISSUE THREAD

**Issue by dpoznik** [astral-sh/ty#2679]
### Summary

When a project defines a namespace package, `ty` cannot resolve imports of `site-packages` packages that use the same namespace. It appears that `ty` finds `./my_namespace/` and looks for `./my_namespace/other_package/`, assuming it must be first-party. Upon not finding it, it reports an `"unresolved-import"`, rather than checking for it under the `site-packages` directory. Python's import system checks both first-party and site-packages for namespace package components, and I believe `ty` should do the same.

Here is an example error message:

```python
error[unresolved-import]: Cannot resolve imported module `ttam.yhaplo.config`
  --> ttam/popgen_utils/haplogroups/trees.py:48:6
   |
47 | from ttam.popgen_utils.helpers.logging_utils import configure_logging
48 | from ttam.yhaplo.config import Config as YhaploConfig
   |      ^^^^^^^^^^^^^^^^^^
49 | from ttam.yhaplo.tree import Tree as YhaploTree
   |
info: Searched in the following paths during module resolution:
info:   1. /Users/dpoznik/projects/ttam.popgen_utils (first-party code)
info:   2. vendored://stdlib (stdlib typeshed stubs vendored by ty)
info:   3. /Users/dpoznik/.pyenv/versions/3.12.7/envs/popgen_utils/lib/python3.12/site-packages (site-packages)
info: make sure your Python environment is properly configured: https://docs.astral.sh/ty/modules/#python-environment
info: rule `unresolved-import` is enabled by default
```
Note that the error message claims to have checked `site-packages` (item 3), but the package `ttam.yhaplo` is installed there and wasn't found, suggesting that `ty` stops searching after finding the first-party `ttam/` directory.

I have a work-around. Adding `site-packages` as an `--extra-search-path` resolves the issue:
```sh
PYTHON_PREFIX=$(python -c "import sys; print(sys.prefix)")
SITE_PACKAGES=$(python -c 'import sysconfig; print(sysconfig.get_path("purelib"))')

ty check --python "${PYTHON_PREFIX}" --extra-search-path "${SITE_PACKAGES}"
```
I don't believe this workaround should be necessary. Rather, I believe `ty` should natively support namespace package merging. Aside: `--python "${PYTHON_PREFIX}"` is necessary here, because I'm using Pyenv-virtualenv, which uses shims rather than setting `VIRTUAL_ENV`.

In case relevant, the namespace packages here are [`pkg_resources`-style namespace packages](https://packaging.python.org/en/latest/guides/packaging-namespace-packages/#pkg-resources-style-namespace-packages), for legacy reasons.

Thanks for this fantastic tool!

### Version

ty 0.0.14 (16597f5e2 2026-01-26)

**Comment by carljm** [comment]
Thanks for the report! I don't think the issue here has anything to do with first-party vs third-party, it's much simpler than that: we currently have no support at all for pkg_resources style namespace packages. We do support the even older `pkgutil.extend_path` style; support for that was added in https://github.com/astral-sh/ruff/pull/20897. So it shouldn't be too hard to support the `pkg_resources` variant as well -- it should just be a matter of supporting detection of both code snippets.

**Comment by carljm** [comment]
(Out of curiosity, are you using another type checker currently which supports this pattern?)

**Comment by carljm** [comment]
@dpoznik if you have the ability to build ty locally, you could test whether https://github.com/astral-sh/ruff/pull/22987 fixes your case.

**Comment by dpoznik** [comment]
> it's much simpler than that: we currently have no support at all for pkg_resources style namespace packages.

Ah, got it. I'm glad I included that detail :)

> So it shouldn't be too hard to support the `pkg_resources` variant as well

Great! I'm glad it's an easy fix!

> (Out of curiosity, are you using another type checker currently which supports this pattern?)

I don't think so. I've been using `mypy`, but I've had `ignore_missing_imports` set to `true` for a whole host of reasons. It was nice to find that, with the work-around, I didn't need to set the `ty` equivalent.

> if you have the ability to build ty locally, you could test whether https://github.com/astral-sh/ruff/pull/22987 fixes your case.

Wow, thanks for the quick fix! I will test this out today and report back.

**Comment by dpoznik** [comment]
The fix works for me!
```sh
[m|ttam.popgen_utils]: ty check --python $(pyenv prefix) | grep Found
Found 8 diagnostics
[m|ttam.popgen_utils]: /Users/dpoznik/repos3p/ruff/target/release/ty version
ty ruff/0.14.14+110 (8d3d8cf95 2026-01-30)
[m|ttam.popgen_utils]: /Users/dpoznik/repos3p/ruff/target/release/ty check --python $(pyenv prefix)
All checks passed!
```

Thank you so much!
